extensions/mssql/src/ui/dialogBase.ts

```diff
@@ -270,28 +277,47 @@ export abstract class DialogBase<DialogResult> {
 		return table;
 	}
 
-	protected addButtonsForTable(table: azdata.TableComponent, addButtonAriaLabel: string, removeButtonAriaLabel: string, addHandler: (button: azdata.ButtonComponent) => Promise<void>, removeHandler: (button: azdata.ButtonComponent) => Promise<void>): azdata.FlexContainer {
+	protected addButtonsForTable(table: azdata.TableComponent, addbuttonComponent: DialogButtonComponent, removeButtonComponent: DialogButtonComponent, editButtonComponent: DialogButtonComponent = undefined): azdata.FlexContainer {
 		let addButton: azdata.ButtonComponent;
+		let editButton: azdata.ButtonComponent;
 		let removeButton: azdata.ButtonComponent;
-		const updateButtons = () => {
+		let buttonComponent: azdata.ButtonComponent[] = [];
+		const updateButtons = (isRemoveEnabled: boolean = undefined) => {
 			this.onFormFieldChange();
-			removeButton.enabled = table.selectedRows?.length === 1 && table.selectedRows[0] !== -1 && table.selectedRows[0] < table.data.length;
+			if (editButtonComponent !== undefined) {
+				editButton.enabled = table.selectedRows?.length === 1 && table.selectedRows[0] !== -1 && table.selectedRows[0] < table.data.length;
```

Comment:
The same enable check is being used for the removeButton below, so they could be consolidated into another local variable to reduce duplication